src/macaron/repo_finder/repo_finder_java.py

```diff
@@ -301,3 +305,46 @@ def _resolve_properties(self, pom: Element, values: list[str]) -> list[str]:
             resolved_values.append(value)
 
         return resolved_values
+
+    @staticmethod
+    def get_deps_dev_info(group: str, artifact: str, version: str) -> dict[str, Any]:
+        """Get additional project information from deps.dev for the given artifact.
+
+        info:
+          - dependentCount (how many other artifacts depend on this)
+          - repo_link (link to the repository)
+          - star_count (number of stars on the repository)
+          - fork_count (number of forks on the repository)
+        """
+        dd_info: dict[str, Any] = {}
+
+        dep_dev_url = f"https://deps.dev/_/s/maven/p/{group}%3A{artifact}/v/{version}"
```

Comment:
I'm curious how this URL was formed. It does not match the typical API or browser usage.

However, I expect using the API is still preferable. The only issue there is that the information required above is only fetchable for repository PURLs: GitHub, etc. We currently use [v3alpha](https://docs.deps.dev/api/v3alpha/#getproject).

Reply:
It's not part of their official API but that's how they populate the project info (star and fork count) at the client side.

Reply:
Added a method to `DepsDepRepoFinder` to get project info from the official API.